packages/svelte/src/lib/Dotlottie.svelte

```diff
@@ -60,114 +72,133 @@
 		if (dotLottieRefCallback) {
 			dotLottieRefCallback(dotLottie);
 		}
-		
+
 		canvas.addEventListener('mouseenter', hoverHandler);
 		canvas.addEventListener('mouseleave', hoverHandler);
 
 		return () => {
-			canvas.removeEventListener('mouseenter', hoverHandler);
-			canvas.removeEventListener('mouseleave', hoverHandler);
-			dotLottie.destroy();
+			if (canvas) {
+				canvas.removeEventListener('mouseenter', hoverHandler);
+				canvas.removeEventListener('mouseleave', hoverHandler);
+			}
+			if (dotLottie) dotLottie.destroy();
 		};
 	});
 
-	$: {
+	$effect(() => {
 		if (dotLottie && typeof layout === 'object') {
 			dotLottie.setLayout(layout);
 		}
-	}
+	});
 
-	$: {
+	$effect(() => {
 		if (dotLottie && typeof marker === 'string') {
 			dotLottie.setMarker(marker);
 		}
-	}
+	});
 
-	$: {
-		if (dotLottie &&  dotLottie.isLoaded && typeof speed == 'number') {
+	$effect(() => {
+		console.log(speed);
```

Comment:
https://github.com/sveltejs/svelte/discussions/14113

Reply:
resolved in https://github.com/LottieFiles/dotlottie-web/pull/391/commits/04873b3f78c00e0c822e1af579e4676eca19a807